Documentation/gettingstarted/dns.rst

```diff
@@ -0,0 +1,203 @@
+
+In this guide we will learn about:
+
+- Controlling egress access to services outside the cluster using DNS-based policies
+- Using patterns (or wildcards) to whitelist a subset of DNS domains
+- Combining DNS, port and API-level rules for restricting external service access
+
+Step 3: Create Example App
+==========================
+
+The file ``dns-sw-app.yaml`` contains ``deathstar``, an internal cluster service and two pods, ``mediabot`` and ``spaceship``. The ``mediabot`` pod needs access to twitter API services for managing the empire's tweets. However, the ``spaceship`` pods shouldn't have any external service access for obvious reasons. Both the pods should be able to access the internal ``deathstar`` service. 
+
+.. parsed-literal::
+
+    $ kubectl create -f \ |SCM_WEB|\/examples/minikube/dns-sw-app.yaml
+    $ kubectl get po,svc
+    NAME                             READY   STATUS    RESTARTS   AGE
+    pod/deathstar-5fc7c7795d-7ljr8   1/1     Running   0          14s
+    pod/deathstar-5fc7c7795d-lvw69   1/1     Running   0          14s
+    pod/mediabot                     1/1     Running   0          14s
+    pod/spaceship                    1/1     Running   0          14s
+
+    NAME                 TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)   AGE
+    service/deathstar    ClusterIP   10.110.127.146   <none>        80/TCP    14s
+    service/kubernetes   ClusterIP   10.96.0.1        <none>        443/TCP   16m
+
+Step 4: Check Current Access
+============================
+
+Since we have not enforced any policies, both ``mediabot`` and ``spaceship`` have access to twitter API service as well as internal ``deathstar`` service. You can confirm this by running the following commands.
+
+.. parsed-literal::
+
+    $ kubectl exec -it mediabot -- curl -sL https://api.twitter.com
+    ...
+
+    $ kubectl exec -it spaceship -- curl -sL https://api.twitter.com
+    ...
+
+    $ kubectl exec -it spaceship -- curl -sL deathstar.default.svc.cluster.local/v1
+    {
+	"name": "Death Star",
+        ...
+        ...
+    }
+
+    $ kubectl exec -it mediabot -- curl -sL deathstar.default.svc.cluster.local/v1
+    {   
+        ...
+        ...
+    }
+
+Step 5: Apply DNS Egress Policy
+===============================
+
+We will enforce a Cilium network policy which allows:
+
+* Only the ``mediabot`` pods have access to ``api.twitter.com``. 
+* Both the pods have access to the internal ``deathstar`` service.
+
+The following Cilium network policy achieves above access controls. 
+
+.. literalinclude:: ../../examples/minikube/dns-policy.yaml
+
+Let's take a closer look at the policy: 
```

Comment:
The policy example you use will have an effect on other services running by including policy elements that apply to all services in the namespaces. You are assuming that only the two services are running in the namespace. The example should be limited to the service you are talking about.

I also see no value in mixing the policy to also include a policy element to whitelists one particular other service. I would whitelist the cluster entity and refer to another section in the doc and state that this example whitelists all egress inside of the cluster for this service, refer to this section to lock it down further.

Otherwise we *require* the user to also tackle service to service policies at the same time

Reply:
In the interest of keeping it simple and focused on DNS policies, removed the internal `deathstar` service and associated policies.